sdks/java/io/kinesis/build.gradle

```diff
@@ -34,6 +34,7 @@ dependencies {
   compile library.java.slf4j_api
   compile library.java.joda_time
   compile library.java.jackson_dataformat_cbor
+  compile library.java.guava
```

Comment:
@aromanenko-dev  As Kenn responded to you in the email, it's good practice to explicitly declare dependency if you're using the class. Without an explicit dependency, it's not clear where the class is coming from (could be 3-hop deep away from your project) and may throw difficult-to-troubleshoot compilation/runtime errors in future when some of the dependencies change.

Reply:
Thank you for fixing this. Don't we need to be aligned with Guava version which is used in `com.amazonaws:amazon-kinesis-producer` in this case? In Beam, we have guava version  "20.0", but KPL depends on version "18.0"

Reply:
No, version 18.0 has not been used.

This is dependency tree of sdks/java/io/kinesis **before** this PR: https://gist.github.com/suztomo/22a5d15e21b4d65507c3be3f38139bb9 . Beam and sdks/java/io/kinesis have been pinning `com.google.guava:guava`'s version to 20.0 through `resolutionStrategy` by [BeamModulePlugin](https://github.com/apache/beam/blob/master/buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy#L1245).

Reply:
Yes, this is what I was concerned about since Amazon KPL depends on Guava 18.0 ([link](https://search.maven.org/artifact/com.amazonaws/amazon-kinesis-producer/0.13.1/jar)). Though, it's a case for other KPL deps as well and it should be ok if it works in the end as expected.

Reply:
@aromanenko-dev the point is that this library has already been pinned to 20.0 for a long time, I think because `BeamModulePlugin` is going in and overriding things even when it doesn't have to. So we know it currently is working and @suztomo is just recording the status quo.

By the way it isn't so simple as KPL depending on 18.0.  This IO module also links against the Kinesis client which depends on 26.0-jre and it has been pinned downward to 20.0. So there's not one version of Guava that these AWS libs use.

Here's the excerpt from the linked dep tree, which is already how it has been for some time.

```
compile - Dependencies for source set 'main' (deprecated, use 'implementation' instead).
+--- com.amazonaws:amazon-kinesis-client:1.13.0
|    +--- com.google.guava:guava:26.0-jre -> 20.0
+--- com.amazonaws:amazon-kinesis-producer:0.13.1
|    +--- com.google.guava:guava:18.0 -> 20.0
```

Highest priority is choosing a version of Guava that works for this connector. So it is OK for it to diverge and use its own version. The aliases at `library.java.*` exist just to help Beam modules cooperate. We probably want to keep moving `library.java.guava` forward, for example I think we need an upgrade support Java 11. If that breaks this module, then we will just have to fork the version.

Is there enough test coverage to catch any problems? Is the Kinesis client library open source so you can clone it and see if it compiles with a new version? That could be a way to check full API compatibility that might be missed by testing.

Reply:
@kennknowles Thank you for details. Yes, there is a known discrepancy between guava versions used in different Kinesis libs and I believe we test it sufficiently with unit and integration tests. Though, we still have a task to move IT into Apache Jenkins env and run it there (the problem that we run it against real AWS instances but it's another question, not elated to this PR).

Back to initial question, to make it clear - I don't insist, I just asked to clarify =) And I agree that since we know that it's working we should not worry too much about that. Kinesis client library is [open source](https://github.com/awslabs/amazon-kinesis-client), so yes, we can check it if needed.

Reply:
So, it's LGTM for KinesisIO from my side